File: src/main/java/com/senderman/anitrackerbot/MongoDBService.java
package com.senderman.anitrackerbot;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.model.Filters;
import org.bson.Document;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class MongoDBService implements DBService {

    private final MongoClient client = MongoClients.create(System.getenv("database"));
    private final MongoDatabase database = client.getDatabase("anime");

    private MongoCollection<Document> user(int id) {
        return database.getCollection(String.valueOf(id));
    }

    @Override
    public void saveAnime(int id, int userId, String url) {
        var doc = new Document("id", id).append("url", url);
        user(userId).insertOne(doc);
    }

    @Override
    public String getAnimeUrl(int id, int userId) {
        return Objects.requireNonNull(user(userId).find(Filters.eq("id", id)).first()).getString("url");
    }

    @Override
    public void deleteAnime(int id, int userId) {
        user(userId).deleteOne(Filters.eq("id", id));
    }

    @Override
    public void dropUser(int userId) {
        user(userId).drop();
    }

    @Override
    public int totalAnimes(int userId) {
        return (int) user(userId).countDocuments();
    }

    @Override
    public boolean idExists(int id, int userId) {
        return user(userId).find(Filters.eq("id", id)).first() != null;
    }

    @Override
    public boolean urlExists(String url, int userId) {
        return user(userId).find(Filters.eq("url", url)).first() != null;
    }

    @Override
    public Map<Integer, String> getAllAnimes(int userId) {
        var user = user(userId).find();
        Map<Integer, String> result = new HashMap<>();
        for (var doc : user) {
            result.put(doc.getInteger("id"), doc.getString("url"));
        }
        return result;
    }
}

File: src/main/java/com/senderman/anitrackerbot/MongoDBService.kt
package com.senderman.anitrackerbot

import com.mongodb.client.MongoCollection
import com.mongodb.client.model.Filters.eq
import com.senderman.MongoClientKeeper
import org.bson.Document
import java.util.*
import kotlin.collections.ArrayList

internal class MongoDBService : DBService {
    private val client = MongoClientKeeper.client
    private val database = client.getDatabase("anime")

    private fun user(id: Int): MongoCollection<Document> = database.getCollection(id.toString())

    override fun saveAnime(id: Int, userId: Int, url: String) {
        val doc = Document("id", id).append("url", url)
        user(userId).insertOne(doc)
    }

    override fun getAnimeUrl(id: Int, userId: Int): String {
        return user(userId).find(eq("id", id)).first()!!.getString("url")
    }

    override fun deleteAnime(id: Int, userId: Int) {
        user(userId).deleteOne(eq("id", id))
    }

    override fun dropUser(userId: Int) {
        user(userId).drop()
    }

    override fun totalAnimes(userId: Int): Int {
        return user(userId).countDocuments().toInt()
    }

    override fun idExists(id: Int, userId: Int): Boolean {
        return user(userId).find(eq("id", id)).first() != null
    }

    override fun urlExists(url: String, userId: Int): Boolean {
        return user(userId).find(eq("url", url)).first() != null
    }

    override fun getAllAnimes(userId: Int): Map<Int, String> {
        val user = user(userId).find()
        val result: MutableMap<Int, String> = HashMap()
        for (doc in user) {
            result[doc.getInteger("id")] = doc.getString("url")
        }
        return result
    }

    override fun getUsersIds(): List<Int> {
        val result = ArrayList<Int>()
        for (userId in database.listCollectionNames()) {
            result.add(userId.toInt())
        }
        return result;
    }
}